typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -285,6 +338,57 @@ export class EvmIsmModule extends HyperlaneModule<
     return updateTxs;
   }
 
+  protected updatePausableIsm({
+    current,
+    target,
+  }: {
+    current: PausableIsmConfig;
+    target: PausableIsmConfig;
+  }): AnnotatedEV5Transaction[] {
+    if (current.paused === target.paused) {
+      return [];
+    }
+
+    const ismInterface = PausableIsm__factory.createInterface();
+    const data = target.paused
+      ? ismInterface.encodeFunctionData('pause')
+      : ismInterface.encodeFunctionData('unpause');
+
+    return [
+      {
+        annotation: `${target.paused ? 'Pausing' : 'Unpausing'} Pausable ISM on chain "${this.chain}" and address "${this.args.addresses.deployedIsm}"`,
+        chainId: this.multiProvider.getEvmChainId(this.chain),
+        to: this.args.addresses.deployedIsm,
+        data,
+      },
+    ];
+  }
+
+  protected updateOffchainLookupIsm({
+    current,
+    target,
+  }: {
+    current: OffchainLookupIsmConfig;
+    target: OffchainLookupIsmConfig;
+  }): AnnotatedEV5Transaction[] {
+    if (setEquality(new Set(target.urls), new Set(current.urls))) {
+      return [];
+    }
+
+    return [
+      {
+        annotation: `Setting urls to ${target.type} ISM on chain "${this.chain}" and address "${this.args.addresses.deployedIsm}"`,
+        chainId: this.multiProvider.getEvmChainId(this.chain),
+        to: this.args.addresses.deployedIsm,
+        // The contract code just replaces the existing array with the new one
+        data: AbstractCcipReadIsm__factory.createInterface().encodeFunctionData(
```

Comment:
Do we maybe want to verify the URL validity before writing to on-chain state? In the Zod schema maybe?

Reply:
sounds like a good idea